### PR TITLE
React App

### PR DESCRIPTION
**Pull request by prakhar070**
React version for the Interview Schedule version.
Things one - 
1. The API backend was modified to include the organizer role along with the functionality of accepting file uploads (for resumes).
2. The front-end is constructed in Reactjs consisting of the pages to create, edit, List the interviews.
3. Asynchronous Fetch API calls are made to fetch data from the backend.
4. JWT is used for the process of authorizing the users of the application.
5. used react hooks for fetching the data for each component.
6. Most of the validations are being checked on the front-end before the data is sent to the backend.
7. The functionality for uploading and fetching resumes from the backend and giving a preview option for the same to the users.

Things that are to be done - 
1. To make use of redux for maintaining the state of our application easily and efficiently.
2. Optimize few areas of application.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import About from './Pages/About';
+import Navbar from './Components/Navbar'
+import Login from './Pages/Login'
+import Signup from './Pages/Signup'
+import {useHistory } from "react-router-dom";
+import {checkLoginStatus} from "./AuthUtils"
+
+
+function App() {
+    let history = useHistory();
```
Comment (joydeep-ib) [inline src/App.js:15]: can be const

Comment (joydeep-ib) [inline src/App.js:15]: Use const

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    
+    const changeStateOfCreateBox = () => { // this will change the state
+        createBox === true ? setCreateBox(false) : setCreateBox(true);
+    }
+    const handleLogin = (data) =>{
+        setLoggedInStatus(true)
+        localStorage.setItem("token", data.access_token);
+        setCurrentUser(data.username);
+        alert("logged-In successdfully");
+        history.push('/interviews')
```
Comment (joydeep-ib) [inline src/App.js:28]: Good

File: src/AuthUtils.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        errors.name = "Please provide a valid string for name that is atleast 2 characters long and must not contain numbers";
+    }
+    if(data.password && data.password_confirmation && data.password_confirmation !== data.password){
+        errors.password = "does not match with confirmed value";
+    }
+    return errors;
+}
+
+const reflectFormErrors = (errors)=>{
+    if(document.getElementById("name") && errors.name)
```
Comment (joydeep-ib) [inline src/AuthUtils.js:30]: Not Advised

File: src/AuthUtils.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        document.getElementById("email").innerText = errors.email
+    if(document.getElementById("password") && errors.password)
+        document.getElementById("password").innerText = errors.password 
+    if(document.getElementById("title") && errors.title)
+        document.getElementById("title").innerText = errors.title 
+    if(document.getElementById("starttime") && errors.starttime)
+        document.getElementById("starttime").innerText = errors.starttime
+    if(document.getElementById("endtime") && errors.endtime)
+        document.getElementById("endtime").innerText = errors.endtime 
+    if(document.getElementById("interviewer") && errors.interviewer)
```
Comment (joydeep-ib) [inline src/AuthUtils.js:42]: use useRef Hook for these stuff

File: src/Components/Form.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        titleRef.value = props.title ? props.title : ""
+        starttimeRef.value = props.starttime ? utcToLocal(props.starttime) : ""
+        endtimeRef.value = props.endtime ? utcToLocal(props.endtime) : ""
+        interviewerRef.value = props.members ? props.members.interviewer.id : ""
+    }
+
+    const collectFormData = () => {
+        let interview = {}
+        interview.id = props.id
+        interview.title = titleRef.value
```
Comment (joydeep-ib) [inline src/Components/Form.js:31]: Use object de-structuring

File: src/AuthUtils.js (hunk 4 above)
Comment (prakhar070) [reply src/AuthUtils.js:42]: Sure, will change that

**Pull request merged**